src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +737,240 @@ impl<
         Ok(())
     }
 
+    /// This function reads veto-related information from the exit-at-rc clarity contract.
+    /// It returns true if the veto succeeded, and false if it failed.
+    pub fn read_veto_state(
```

Comment:
why is `self` mutable if this is a "read"?

Reply:
`with_read_only_clarity_tx` requires `&mut self`, since the MARF operations all require `&mut self`, since the `TrieFileStorage` keeps mutable benchmarking data on the read path.